include/npcomp/Dialect/Torch/IR/TorchBase.td

```diff
@@ -0,0 +1,31 @@
+//===-------------------------------------------------------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef TORCH_BASE
+#define TORCH_BASE
+
+include "mlir/IR/OpBase.td"
+
+def Torch_Dialect : Dialect {
+  let name = "torch";
+  let cppNamespace = "::mlir::NPCOMP::Torch";
+  let description = [{
+    Top-level dialect for interfacing PyTorch and MLIR.
+
+    This dialect contains types and structural ops that model enough of
+    PyTorch's behavior to allow for easy import/call-out. While not aiming to
+    be completely isomorphic, it is laid out to make conversion in/out
+    systematic for the supported features.
```

Comment:
Do we have enough clarity on the "supported features" to discuss them, even aspirationally here? I tend to find that we never come back to these descriptions, so it pays to have something a bit forward looking. Don't have to be added now. Just whenever you have a jolt of clarity please do add here.

See e.g. the TCF dialect: https://github.com/llvm/mlir-npcomp/blob/fb895173f291afaaa9a4cb5a80daa6d0983dc169/include/npcomp/Dialect/TCF/IR/TCFBase.td#L18